src/ast.rs

```diff
@@ -455,6 +455,31 @@ impl Statement {
     }
 }
 
+// The function will take a vector of statements and reduce them down
+// to only their function calls.
```

Comment:
Probably should expand this comment to call out the fact that it's all possible function calls.  If I write something like:

```
if false {
    some_call(...);
}
```

Then this will still include some_call() even though it's known to be unreachable.  That's certainly the behavior we want here, but probably worth noting in the comment for future reference.